panic if I specify a non existing envelope

The panic

```
Downloading esp32cfoo firmware from https://github.com/toitlang/envelopes/releases/download/v2.0.0-alpha.152/firmware-esp32cfoo.envelope.gz ...
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0x1346f13]

goroutine 1 [running]:
github.com/toitlang/jaguar/cmd/jag/commands.downloadGzipped({0x1646a88?, 0xc0001f2f60?}, {0xc000040c40?, 0x14ea085?}, {0xc00009fbc0, 0x56})
        D:/a/jaguar/jaguar/cmd/jag/commands/envelope.go:44 +0x53
github.com/toitlang/jaguar/cmd/jag/commands.downloadPublishedFirmware({0x1646a88, 0xc0001f2f60}, {0xc00002f338, 0x10}, {0xc0000360f0, 0x9})
        D:/a/jaguar/jaguar/cmd/jag/commands/envelope.go:136 +0x2ac
github.com/toitlang/jaguar/cmd/jag/commands.GetCachedFirmwareEnvelopePath({0x1646a88, 0xc0001f2f60}, {0xc00002f338, 0x10}, {0xc0000360f0, 0x9})
        D:/a/jaguar/jaguar/cmd/jag/commands/envelope.go:27 +0x110
github.com/toitlang/jaguar/cmd/jag/commands.FlashCmd.func1(0xc000228c00, {0xc0000c5880, 0x0, 0x8?})
        D:/a/jaguar/jaguar/cmd/jag/commands/flash.go:97 +0x5ea
github.com/spf13/cobra.(*Command).execute(0xc000228c00, {0xc0000c5800, 0x8, 0x8})
        C:/Users/runneradmin/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:983 +0xab4
github.com/spf13/cobra.(*Command).ExecuteC(0xc000222300)
        C:/Users/runneradmin/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1115 +0x425
github.com/spf13/cobra.(*Command).Execute(...)
        C:/Users/runneradmin/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1039
github.com/spf13/cobra.(*Command).ExecuteContext(...)
        C:/Users/runneradmin/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1032
main.main()
        D:/a/jaguar/jaguar/cmd/jag/main.go:34 +0x1bb
```

jag verion

```
jag version
Version:         v1.37.2
SDK version:     v2.0.0-alpha.152
Build date:      2024-06-04T10:29:11Z
```

I'd generally expect no panic, and some other error message instead

--- a/cmd/jag/commands/envelope.go
+++ b/cmd/jag/commands/envelope.go
@@ -29,10 +29,10 @@ func GetCachedFirmwareEnvelopePath(ctx context.Context, version string, model st
 		if err := downloadPublishedFirmware(ctx, version, model); err != nil {
 			fmt.Printf("Failed to download firmware: %v\n", err)
 			switch model {
-			case "esp32", "esp32s3", "esp32c3", "esp32s2":
+			case "esp32", "esp32c3", "esp32c6", "esp32s2", "esp32s3":
 				// These names are correct. Simply return the error.
-			case "ESP32-S3", "ESP32-C3", "ESP32-S2":
-				fmt.Printf("Chip model names must be lowercase without dashes. Please try again with 'esp32s3', 'esp32c3', or 'esp32s2'.\n")
+			case "ESP32-C3", "ESP32-C6", "ESP32-S2", "ESP32-S3":
+				fmt.Printf("Chip model names must be lowercase without dashes. Please try again with 'esp32c3', 'esp32c6', 'esp32s2', or 'esp32s3'.\n")
 			default:
 				fmt.Printf("Make sure the model name is correct. You can find supported models at\n")
 				fmt.Printf("https://github.com/toitlang/envelopes/releases/tag/%s.\n", version)
@@ -55,6 +55,9 @@ func downloadGzipped(ctx context.Context, url string, path string) error {
 	bundle, err := download(ctx, url)
 	if err != nil {
 		return err
+	}
+	if bundle == nil {
+		return fmt.Errorf("failed to download firmware")
 	}
 	defer bundle.Close()
 	return storeGzipped(bundle, path)

--- a/cmd/jag/commands/util.go
+++ b/cmd/jag/commands/util.go
@@ -574,6 +574,10 @@ func download(ctx context.Context, url string) (io.ReadCloser, error) {
 		return nil, fmt.Errorf("failed downloading from %s: %v", url, resp.Status)
 	}
 
+	if resp.Body == nil {
+		return nil, fmt.Errorf("received nil response body from %s", url)
+	}
+
 	progress := pb.New64(resp.ContentLength)
 	return progress.Start().NewProxyReader(resp.Body), nil
 }

